internal/app/replication/mysql.go

```diff
@@ -0,0 +1,151 @@
+package replication
+
+import (
+	"database/sql"
+
+	"github.com/yandex/mysync/internal/log"
+	"github.com/yandex/mysync/internal/mysql"
+)
+
+type ExternalReplication struct {
+	UnimplementedExternalReplication
+	IsExternalReplicationSupported bool
+	logger                         *log.Logger
+}
+
+func NewExternalReplication(IsExternalReplicationSupported bool, logger *log.Logger) (*ExternalReplication, error) {
+	return &ExternalReplication{
+		IsExternalReplicationSupported: IsExternalReplicationSupported,
+		logger:                         logger,
+	}, nil
+}
```

Comment:
Here we should return ExternalReplication{} or UnimplementedExternalReplication{} depending on IsExternalReplicationSupported setting.